lib/extensions/descendant_loader.rb

```diff
@@ -241,6 +241,26 @@ def clear_class_inheritance_relationships
   include Cache
   include Mapper
 
+  EXCLUDED_PARENTS = %w[
+    ApplicationRecord
+    StandardError
+    ApplicationController
+    Api::BaseController
+    BaseController
+    MiqDecorator
+    ApplicationHelper::Toolbar::Override
+    ApplicationHelper::Button::Basic
+    ApplicationHelper::Toolbar::Basic
+    ApplicationHelper::Button::ButtonNewDiscover
+    TreeBuilder
+    TreeNode::Node
+    Node
+  ]
+
+  def excluded_parent?(parent)
+    EXCLUDED_PARENTS.include?(parent.to_s)
+  end
```

Comment:
Ok, I've split this excluded parents concept into it's own commit.  We can choose to drop it as it's only an optimization.  We can followup with it if it's questionable.

Reply:
This isn't needed.  @kbrock helped me find out that if the `__update_callbacks` entrypoint for `descendants` is excluded, there is no time that we are calling `descendants` on these high level classes.  Dropping this commit.